src/utils/wallet.js

```diff
@@ -101,6 +103,55 @@ export class Wallet {
       return await this.getAccount(this.accountId).getAccessKeys()
    }
 
+   async connectWapm(wamp) {
+      try {
+         return await new Promise((resolve, reject) => {
+            wamp.onopen = session => resolve(session)
+            wamp.onclose = reason => reject(reason)
+            wamp.open()
+         });
+      } catch (error) {
+         console.error('Connection failure due to:', error)
+         return
+      }
+   }
+
+   async getTransactions(accountId = '') {
+      if (!this.accountId) return null
+      if (!accountId) accountId = this.accountId
+
+      const wamp = new autobahn.Connection({
+         realm: 'near-explorer',
+         transports: [
+            {
+               url: WAMP_NEAR_EXPLORER_URL,
+               type: 'websocket'
+            }
+         ],
+         retry_if_unreachable: true,
+         max_retries: Number.MAX_SAFE_INTEGER,
+         max_retry_delay: 10
+      })
+
+      const wampSession = await this.connectWapm(wamp)
+      if (!wampSession) return
+
+      try {
+         return await wampSession.call(
+            'com.nearprotocol.testnet.explorer.select',
+            [
+               `SELECT * FROM transactions WHERE signer_id = :accountId OR receiver_id = :accountId`,
```

Comment:
Oh, let's LIMIT this with some sane number, and you may also need ORDER BY (well, to do that, you would need block.height, so JOIN is also required).

```js
`SELECT transactions.hash, transactions.signer_id as signerId, transactions.receiver_id as receiverId, transactions.actions, transactions.block_hash as blockHash, blocks.timestamp as blockTimestamp
          FROM transactions
          LEFT JOIN blocks ON blocks.hash = transactions.block_hash
          WHERE signer_id = :accountId OR receiver_id = :accountId
          ORDER BY blocks.height DESC
          LIMIT :limit`,
{ accountId, limit: 10 }
```

Reply:
cool